tests/main/apparmor-prompting-integration-tests/task.yaml

```diff
@@ -0,0 +1,100 @@
+summary: Check that AppArmor Prompting works end-to-end
+
+details: |
+    Test that the AppArmor Prompting subsystems work by simulating common usage
+    scenarios. Manually carry out operations which generate prompts and, using
+    the prompting-client snap in scripted mode, check that those prompts match
+    what is expected, and that the client suffessfully allows the operations.
+
+systems:
+  - ubuntu-16.04-*
+  - ubuntu-18.04-64
+  - ubuntu-2*
+  - ubuntu-core-*
+
+environment:
+    VARIANT/read_single_allow: read_single_allow
+    VARIANT/read_single_deny: read_single_deny
+    VARIANT/write_single_allow: write_single_allow
+    VARIANT/write_single_deny: write_single_deny
+    VARIANT/create_multiple_allow: create_multiple_allow
+    VARIANT/create_multiple_deny: create_multiple_deny
+    VARIANT/download_file_defaults: download_file_defaults
+    VARIANT/download_file_safer: download_file_safer
+    VARIANT/download_file_conflict: download_file_conflict
+    VARIANT/timespan_allow: timespan_allow
+    VARIANT/timespan_deny: timespan_deny
+    VARIANT/create_multiple_actioned_by_other_pid_always_allow: create_multiple_actioned_by_other_pid_always_allow
+    VARIANT/create_multiple_actioned_by_other_pid_always_deny: create_multiple_actioned_by_other_pid_always_deny
+    VARIANT/create_multiple_not_actioned_by_other_pid_single_allow: create_multiple_not_actioned_by_other_pid_single_allow
+    VARIANT/create_multiple_not_actioned_by_other_pid_single_deny: create_multiple_not_actioned_by_other_pid_single_deny
+    #VARIANT/overwrite_file: overwrite_file
+
+prepare: |
+    tests.session prepare -u test
+    #shellcheck disable=SC2016
+    tests.session -u test exec sh -c 'mkdir -p "$HOME/integration-tests"' # $HOME must be that of test user
+    snap install prompting-client
+
+restore: |
+    #shellcheck disable=SC2016
+    tests.session -u test exec sh -c 'rm -rf "$HOME/integration-tests"' # $HOME must be that of test user
+    tests.session restore -u test
+
+debug: |
+
+execute: |
+    echo "Enable prompting via snap client where possible"
+    # Prompting is unsupported everywhere but the Ubuntu non-core systems with
+    # kernels which support apparmor prompting
+    if ! os.query is-ubuntu || os.query is-core || ! grep 'prompt' /sys/kernel/security/apparmor/features/policy/permstable32 ; then
```

Comment:
Any Ubuntu kernel new enough to support the `prompt` permission should also support prompting. If a kernel doesn't because `features/policy/notify/` exists but doesn't contain a file called `user` with `file` in it, then something unexpected has happened in kernel land and we'd like to know.

The checks for `features/policy/notify/user` are because some future systems (non-Ubuntu, is my understanding) will use the same notify socket for the AppArmor complain redirect as is used for prompt messages, so we need a more robust check in snapd if that system is in place but prompting isn't supported. Any Ubuntu system prior to 24.10 does not have `features/policy/notify/` though, and prompting support is assumed by the presence of the `prompt` permission and the notify socket.

Basically, snapd in production is extra strict since it may run on systems with Ubuntu kernels, upstream kernels, custom kernels, etc., but in spread tests on Ubuntu, all systems with a new enough kernel (including jammy since it uses 6.5 in google cloud images iirc) should support prompting and we want to test them to catch if anything out of the ordinary happens.

(Sorry for the long-winded answer, writing this out helps me remind myself of the situation)